src/meta/meta_ckpt.c

```diff
@@ -835,11 +836,14 @@ __wt_meta_ckptlist_get(
             WT_ERR(ret);
         }
     } else {
-        WT_ERR(__wt_metadata_search(session, fname, &config));
+        WT_ERR(__wt_btree_shared_base_name(session, &fname, NULL, &name_buf));
+        WT_ERR(__wt_metadata_search(session, name_buf == NULL ? fname : name_buf->data, &config));
```

Comment:
What is this actually changing? As far as I can tell, `__wt_btree_shared_base_name` should set `fname` and `name_buf` to effectively the same thing, i.e. the line `*namep = (const char *)name_buf->data;`.

Reply:
It truncates the checkpoint id part of name/checkpointid

Reply:
Oh I see - this is actually kind of subtle. If there isn't a trailing slash, `__wt_btree_shared_base_name` will leave `fname` untouched and leave `name_buf` as NULL, so in this case it searches for the same thing as the old code.